### PR TITLE
webuipoc: Add Eslint

### PR DESCRIPTION
**Pull request by Moeez905**

### CODE REVIEW

File: addOns/webuipoc/webuipoc.gradle.kts [hunk 1]
```diff
@@ -41,6 +41,19 @@ for (dir in pocsSrcDir.listFiles()!!) {
                 }
             },
         )
+
+if (normalizedPocName == "ReactWebUI") {
++            val lintTask =
```
Comment (thc202) [inline addOns/webuipoc/webuipoc.gradle.kts:46]: The `+` should be removed, that's part of the diff syntax.

File: addOns/webuipoc/src/main/pocs/reactWebUI/src/App.js [hunk 2]
```diff
@@ -2,9 +2,11 @@
 import './App.css';
```
Comment (thc202) [inline addOns/webuipoc/src/main/pocs/reactWebUI/src/App.js:2]: These changes should be removed form this PR.

File: addOns/webuipoc/webuipoc.gradle.kts (hunk 1 above)
Comment (Moeez905) [reply addOns/webuipoc/webuipoc.gradle.kts]: done

File: addOns/webuipoc/src/main/pocs/reactWebUI/src/App.js (hunk 2 above)
Comment (thc202) [reply addOns/webuipoc/src/main/pocs/reactWebUI/src/App.js]: To be clear, I mean the whole UI changes that are already being done in the other PR.

File: addOns/webuipoc/webuipoc.gradle.kts (hunk 1 above)
Comment (thc202) [reply addOns/webuipoc/webuipoc.gradle.kts]: You might need to run the Gradle `spotlessApply` task.

Comment (Moeez905) [reply addOns/webuipoc/webuipoc.gradle.kts]: applied

File: addOns/webuipoc/src/main/pocs/reactWebUI/package.json [hunk 3]
```diff
@@ -37,6 +37,10 @@
     ]
   },
   "devDependencies": {
+    "@eslint/js": "^9.4.0",
+    "eslint": "^8.57.0",
```
Comment (thc202) [inline addOns/webuipoc/src/main/pocs/reactWebUI/package.json:41]: We are back to two major versions, if v8 is still a must (jsx-eslint/eslint-plugin-react#3699) we should not add v9 and keep v8.

**Comment by thc202** [comment]
This needs to be added back: https://github.com/zaproxy/zap-extensions/pull/5460#discussion_r1638216608

File: addOns/webuipoc/src/main/pocs/reactWebUI/package.json (hunk 3 above)
Comment (Moeez905) [reply addOns/webuipoc/src/main/pocs/reactWebUI/package.json:41]: it comes by default while integrating Eslint, Removing it

**Comment by Moeez905** [comment]
> This needs to be added back: [#5460 (comment)](https://github.com/zaproxy/zap-extensions/pull/5460#discussion_r1638216608)

added

File: addOns/webuipoc/src/main/pocs/reactWebUI/package.json
Comment (thc202) [inline addOns/webuipoc/src/main/pocs/reactWebUI/package.json:1]: - Missing the lint script entry.
 - Is `eslintConfig` still needed?

File: addOns/webuipoc/webuipoc.gradle.kts [hunk 4]
```diff
@@ -41,7 +41,19 @@ for (dir in pocsSrcDir.listFiles()!!) {
...
+                    group = pocBuildTasksGroup
+                    dependsOn(installTask)
+                    workingDir = dir
+                    args.set(arrayListOf("run", "lint"))
+                }
+            tasks.named(LifecycleBasePlugin.CHECK_TASK_NAME) {
+                dependsOn(lintTask)
+            }
+        }
+    	} else {
```
Comment (thc202) [inline addOns/webuipoc/webuipoc.gradle.kts:56]: Needs `spotlessApply`.

File: addOns/webuipoc/src/main/pocs/reactWebUI/eslint.config.mjs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import globals from "globals";
```
Comment (thc202) [inline addOns/webuipoc/src/main/pocs/reactWebUI/eslint.config.mjs:1]: Does this config format apply to v8?

Comment (Moeez905) [reply addOns/webuipoc/src/main/pocs/reactWebUI/eslint.config.mjs:1]: yes

File: addOns/webuipoc/src/main/pocs/reactWebUI/package.json
Comment (Moeez905) [reply addOns/webuipoc/src/main/pocs/reactWebUI/package.json:1]: I have just ran the default eslint setup script that is included with eslint package. No manual config is done.

Comment (thc202) [reply addOns/webuipoc/src/main/pocs/reactWebUI/package.json:1]: The `eslintConfig` was already in the file (even before eslint was being added).

Comment (thc202) [reply addOns/webuipoc/src/main/pocs/reactWebUI/package.json:1]: The first point still needs to be addressed in any case.

**Comment by thc202** [comment]
The branch has conflicts.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**